modules/tests/test_opendaq_device_modules/test_streaming.cpp

```diff
@@ -576,36 +588,40 @@ TEST_F(NativeDeviceStreamingTest, ChangedDataDescriptorBeforeSubscribeNativeDevi
 
     SignalConfigPtr serverSignalPtr = serverInstance.getSignalsRecursive(search::LocalId("ByteStep"))[0];
     MirroredSignalConfigPtr clientSignalPtr = clientInstance.getSignalsRecursive(search::LocalId("ByteStep"))[0];
-    
-    clientInstance.getDevices()[0].setPropertyValue("ChangeDescriptors", 1);
-    ASSERT_EQ(callCount, sigCount);
 
-    const DataDescriptorPtr valueDataDesc = serverSignalPtr.getDescriptor();
-    const DataDescriptorPtr domainDataDesc = serverSignalPtr.getDomainSignal().getDescriptor();
+    for (int i = 1; i < 5; ++i)
+    {
+        callCount = 0;
+        clientInstance.getDevices()[0].setPropertyValue("ChangeDescriptors", i % 2);
```

Comment:
Why is this property mechanism used rather than direct modifying of the server signal descriptor in the same manner as in the test fixture above ? Was it introduced to avoid some side effects of the NativeConfig protocol ?

Reply:
It's because the native protocol is push-based and we cannot configure the descriptors from the client side. When we change the descriptors on the server, the client is only after a while notified of the change. This simulates a change happening on the client that results in a descriptor change, with an immediate subscription afterwards.

If using the above method, we'd need to wait for the descriptor to change on the client side as well.